### ISSUE TITLE
[FeG][Gy] CCA-U with result code 5002 UNKNOWN SESSION ID and 5004 INVALID AVP VALUE after Re-Auth Process

### ISSUE THREAD

**Issue by lulloaX** [magma/magma#8663]
### Your Environment

- **Version:** 1.6
- **Affected Component:** Fed Gateway
- **Affected Subcomponent:** Fed Gateway
- **Deployment Environment:** local docker-compose (orc8r), Openstack AGW/FeG)

### Describe the Issue

FeG received a command code: 258 Re-Auth from Commercial OCS with Re-Auth Request Type AVP on 0 value (AUTHORIZE_ONLY) in Frame #125867.
![image](https://user-images.githubusercontent.com/28659806/129771890-3a8f2bf9-b0fa-4d0c-8caa-d33a697cddd1.png)
 
FeG replied it with result code 2002 DIAMETER_LIMITED_SUCCESS.
![image](https://user-images.githubusercontent.com/28659806/129772061-978cdd4f-f813-41bf-b718-25bddbcf8084.png)

After that FeG send three CCR-U (one per each MSCC) with 3GPP-Reporting-Reason AVP on 7 value (FORCED_AUTHORIZATION), but commercial OCS replied CCA-U messages with result code 5002 DIAMETER_UNKNOW_SESSION ID and 5004 DIAMETER_INVALID_AVP_VALUE, it forced an CCR-T from FeG to OCS but it also failed with result code 5002 DIAMETER_UNKNOW_SESSION_ID on CCA-T.
![image](https://user-images.githubusercontent.com/28659806/129774933-218d9882-9bf8-46c8-9bd4-1278f0e04fc7.png)

**Expected behavior**

Because of OCS is on commercial state, the same Re-Auth procedure happen everyday on commercial scenario but in this case SPGW send a CCR-U message for first MSCC and wait its CCA-U before send the second CCR-U for next MSCC.

**Additional context**

- Attach the logs for session_proxy (Line 4887) and s6a_proxy (Line 2112) services.
[fegActive_1708_logs.zip](https://github.com/magma/magma/files/7002125/fegActive_1708_logs.zip)

- Attach the pcap trace, see Frame #125867
[trace_1608_feg03_act.zip](https://github.com/magma/magma/files/7002128/trace_1608_feg03_act.zip)

**Comment by uri200** [comment]
Hey @LuisUlloaB the pcap provided doesn't have frame 125867 (RAR)

Can you please provide pcap for that?

Can you please split this into two issues (RAR and CCR-U)

**Comment by uri200** [comment]
@LuisUlloaB can you repeat the test with grpc logging enabled on feg please 
`MAGMA_PRINT_GRPC_PAYLOAD: 1`
`-v=9`

```
 session_proxy:
    <<: *goservice
    environment:
      USE_GY_FOR_AUTH_ONLY: ${USE_GY_FOR_AUTH_ONLY}
      GY_SUPPORTED_VENDOR_IDS: ${GY_SUPPORTED_VENDOR_IDS}
      GY_SERVICE_CONTEXT_ID: ${GY_SERVICE_CONTEXT_ID}
      DISABLE_REQUESTED_SERVICE_UNIT_AVP: ${DISABLE_REQUESTED_SERVICE_UNIT_AVP}
      MAGMA_PRINT_GRPC_PAYLOAD: 1             <---------
    container_name: session_proxy
    command: envdir /var/opt/magma/envdir /var/opt/magma/bin/session_proxy -logtostderr=true -v=9   <-----
```

Also can you include sessiond logs with debug and grpc enabled: on `sessiond.yml` change 
```
log_level: DEBUG
print_grpc_payload: true
```

**Comment by lulloaX** [comment]
Dear @uri200 please for Gy interface on .pcap file filter using `tcp.port==5112` , then use `Analyze > Decode` option as bellow image.
![image](https://user-images.githubusercontent.com/28659806/130537395-cd55fd90-d3c2-4f09-9342-7c91e349eccf.png)

After that you will see RAR message on frame #125867.
![image](https://user-images.githubusercontent.com/28659806/130537498-0242678c-f44e-4754-84c1-3c7fd3e28955.png)

**Comment by uri200** [comment]
Hey @LuisUlloaB I found the pcap, thanks. What we are missing is session_proxy grpc logs and sessiond logs to be able to trace the call.

**Comment by lulloaX** [comment]
Hi @uri200 we enabled the grpc option for `session_proxy` and `sessiond`. Find attached the logs for session_proxy, syslog (include sessiond logs) and new .pcap trace.

- Session_Proxy:
[session_proxy_2308.zip](https://github.com/magma/magma/files/7039729/session_proxy_2308.zip)

- .PCAP trace:
[trace_feg2308.zip](https://github.com/magma/magma/files/7039732/trace_feg2308.zip)

- Syslog
[syslog_2308.tar.zip](https://github.com/magma/magma/files/7039782/syslog_2308.tar.zip)

**Comment by uri200** [comment]
Hey @LuisUlloaB , I can see `diameter-success` on the pcap. 

In case more logs were required, please start capturing all the log (for example `sudo journalctl -fu magma@sessiond > sessiond.log`) and then run the test. All logs should be collected at the same time so we can compare between them.

With the log files, please add a description of what the test was doing so we can have the proper context. Add the ports you are using for HSS, GX and GY so we can decode all messages.

Note syslog you sent is 4gb file which we can't analyze

**Comment by uri200** [comment]
@LuisUlloaB I was able to filter out sessiond logs from syslog. 
Can you tell me what is the issue you are observing now?

**Comment by lulloaX** [comment]
@uri200  sorry my friend, we will consider journalctl option on next feedbacks. About ports, we are using these:
- HSS:   SCTP  port: 7002
- Gx:     TCP    port: 5111
- Gy:     TCP    port: 5112

The issue we are facing is on Gy interface:
1. OCS send Re-Auth Request (RAR), but it is replied by FeG with result code 2002 (DIAMETER_LIMITED_SUCCED).
2. Then FeG send three CCR-U messages (one per MSCC) but only the last one result success by OCS. Two first obtained result codes 5004 and 5002.
3. After these failed CCR-U session is terminated by FeG.
All this sequence is captured on last .pcap I send, just filter it using `((tcp.port==5112 and diameter) ) && !(diameter.cmd.code == 280)`
![image](https://user-images.githubusercontent.com/28659806/130662745-edeb74e3-ad54-431e-b756-f0e59965c880.png)

We already compared issue trace and commercial trace on Gy interface, found the difference is commercial SPGW send a CCR-U message for first MSCC and wait its CCA-U before send the second CCR-U for next MSCC. Current commercial OCS is Ericcson so its very difficult perform changes there, maybe we can change this behavior directly on FeG.
Here is a picture of Gy commercial trace.
![image](https://user-images.githubusercontent.com/28659806/130664672-ddf4d729-297c-40e4-ab75-61549fa4bf80.png)

We talked about this issue with @joary and Emerson Acuna, they have knowlegde about it.

**Comment by uri200** [comment]
Hey @LuisUlloaB , `DIAMETER_LIMITED_SUCCED` is fine. Per our code that means the update was initiated but not completed yet. The CCR-U happens few seconds later.

For the second issue, Magma sends the requests in parallel. OCS should be able to handle that. We can't wait for each CCR-U to come. That would not work once you have several updates to send. So this seems to be an issue on OCS. Can you repeat the test using two two services (RG 201 + SI 201 and RG 200 + SI 200,  do not use RG 201 + SI 375) 

The session is terminated because OCS responds back with a 5002 and 5004 codes which are `permanent errors`. Those cause the termination.

**Comment by uri200** [comment]
This PR has a preliminar solution for that issue
https://github.com/magma/magma/pull/9055
